sunpy/coordinates/ephemeris.py

```diff
@@ -282,46 +287,97 @@ def get_horizons_coord(body, time='now', id_type=None, *, include_velocity=False
     ...                    time={{'start': '2020-12-01',
     ...                           'stop': '2020-12-02',
     ...                           'step': '12'}})  # doctest: +REMOTE_DATA
-    INFO: Obtained JPL HORIZONS location for Solar Orbiter (spacecraft) (-144 [sunpy.coordinates.ephemeris]
+    INFO: Obtained JPL HORIZONS location for Solar Orbiter (spacecraft) (-144) [sunpy.coordinates.ephemeris]
     ...
     """
-    # Import here so that astroquery is not a module-level dependency
-    import astroquery
-    from astroquery.jplhorizons import Horizons
-
-    if id_type is None and version.parse(astroquery.__version__) < version.parse('0.4.4'):
-        # For older versions of astroquery retain default behaviour of this function
-        # if id_type isn't manually specified.
-        id_type = 'majorbody'
+    args = {
+        'EPHEM_TYPE': 'VECTORS',
+        'OUT_UNITS': 'AU-D',
+        'CENTER': '500@10',
+        'CSV_FORMAT': 'YES',
+    }
+
+    if id_type in [None, 'smallbody', 'designation', 'name', 'asteroid_name', 'comet_name']:
+        prepend = {
+            None: "",
+            'smallbody': "",
+            'designation': "DES=",
+            'name': "NAME=",
+            'asteroid_name': "ASTNAM=",
+            'comet_name': "COMNAM=",
+        }
+        if id_type == 'smallbody' and body[-1] != ';':
+            body += ';'
+        args['COMMAND'] = f"'{prepend[id_type]}{body}'"
+    else:
+        raise ValueError("Invalid id_type")
 
     if isinstance(time, dict):
         if set(time.keys()) != set(['start', 'stop', 'step']):
             raise ValueError('time dictionary must have the keys ["start", "stop", "step"]')
-        epochs = time
-        jpl_fmt = '%Y-%m-%d %H:%M:%S.%f'
-        epochs['start'] = parse_time(epochs['start']).tdb.strftime(jpl_fmt)
-        epochs['stop'] = parse_time(epochs['stop']).tdb.strftime(jpl_fmt)
+        jpl_fmt = "'%Y-%m-%d %H:%M:%S.%f'"
+        args['START_TIME'] = parse_time(time['start']).tdb.strftime(jpl_fmt)
+        args['STOP_TIME'] = parse_time(time['stop']).tdb.strftime(jpl_fmt)
+        args['STEP_SIZE'] = time['step']
     else:
         obstime = parse_time(time)
+        if obstime.size >= 10000:
+            raise ValueError("For more than 10,000 time values, use dictionary input.")
         array_time = np.reshape(obstime, (-1,))  # Convert to an array, even if scalar
         epochs = array_time.tdb.jd.tolist()  # Time must be provided in JD TDB
+        args['TLIST'] = '\n'.join(str(epoch) for epoch in epochs)
+        args['TLIST_TYPE'] = 'JD'
+
+    with tempfile.TemporaryFile('w+t') as query:
+        contents = "!$$SOF\n" + '\n'.join(f"{k}={v}" for k, v in args.items())
+        log.debug(f"File-based JPL HORIZONS query:\n{contents}")
+
+        query.write(contents)
+        query.seek(0)
```

Comment:
Is it necessary to write this to disk, or put another way is there a benefit? I'm pretty sure you could pass `contents` directly to request as post data, might have to wrap in StringIO or something first.

Reply:
Heh, whoops, indeed you *can* simply provide a string.  The `requests` documentation isn't particularly clear about all the different forms of data it can accept...